### PR TITLE
[opt](binlog) Log table rename when reply

### PR DESCRIPTION
**Pull request by lsy3993**
## Proposed changes

Issue Number: close #xxx

Related PR is : #37031 , when replay rename table, addTableRename is needed

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 1]
```diff
@@ -4690,6 +4690,10 @@ public void replayRenameTable(TableInfo tableInfo) throws MetaNotFoundException
                 db.unregisterTable(tableName);
                 table.setName(newTableName);
                 db.registerTable(table);
+
+                TableInfo newTableInfo = TableInfo.createForTableRename(db.getId(), table.getId(), tableName,
+                    newTableName);
+                editLog.logTableRename(newTableInfo);
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:4696]: We should not write edit log when replaying .

**Review by dataroaring — CHANGES_REQUESTED** [review]

**Comment by dataroaring** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 2]
```diff
@@ -4690,6 +4690,8 @@ public void replayRenameTable(TableInfo tableInfo) throws MetaNotFoundException
                 db.unregisterTable(tableName);
                 table.setName(newTableName);
                 db.registerTable(table);
+
+                Env.getCurrentEnv().getBinlogManager().addTableRename(tableInfo, logId);
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:4694]: move this line to EditLog.java:308

**Review by w41ter — CHANGES_REQUESTED** [review]

**Review by w41ter — CHANGES_REQUESTED** [review]
Please add `addTableRename` to the replay of OP_RENAME_ROLLUP and OP_RENAME_PARTITION too.

**Review by w41ter — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Comment by lsy3993** [comment]
run buildall

**Comment by lsy3993** [comment]
run buildall

**Comment by lsy3993** [comment]
run buildall

**Comment by lsy3993** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**